inc/azure_c_shared_utility/xlogging.h

```diff
@@ -109,6 +116,53 @@ typedef void(*LOGGER_LOG_GETLASTERROR)(const char* file, const char* func, int l
+        (void)printf(format, __VA_ARGS__);                                                                                                                               \
+    }                                                                                                                                                                    \
+    {                                                                                                                                                                    \
+        LOGGER_LOG l = xlogging_get_log_function();                                                                                                                      \
+        if (l != NULL)                                                                                                                                                   \
+        {                                                                                                                                                                \
+            if(log_category == AZ_LOG_ERROR)                                                                                                                             \
+            {                                                                                                                                                            \
+                char* stackAsString = getStackAsString();                                                                                                                \
+                if (stackAsString == NULL)                                                                                                                               \
+                {                                                                                                                                                        \
+                    l(log_category, __FILE__, FUNC_NAME, __LINE__, log_options, format, __VA_ARGS__);                                                                    \
+                }                                                                                                                                                        \
+                else                                                                                                                                                     \
+                {                                                                                                                                                        \
+                    size_t formatSize = strlen(format);                                                                                                                  \
+                    char* formatWithStack = (char*)malloc(formatSize + sizeof("STACK_PRINT_FORMAT"));                                                                    \
```

Comment:
>malloc [](start = 51, length = 6)

This needs to be logging_malloc

Reply:
Changed

---
In reply to: [276783533](https://github.com/Azure/azure-c-shared-utility/pull/302#discussion_r276783533) [](ancestors = 276783533)